src/test/java/seedu/pill/command/HelpCommandTest.java

```diff
@@ -0,0 +1,168 @@
+        ItemMap itemMap = new ItemMap();
+        Storage storage = new Storage();
+
+        helpCommand.execute(itemMap, storage);
+
+        String output = outContent.toString().trim();
+        assertTrue(output.contains("Available commands:"), "Should show general help for empty string input");
+    }
+
+    @Test
+    public void testHelpWithWhitespaceOnly() throws PillException {
+        HelpCommand helpCommand = new HelpCommand("   ", false);
+        ItemMap itemMap = new ItemMap();
+        Storage storage = new Storage();
+
+        helpCommand.execute(itemMap, storage);
+
+        String output = outContent.toString().trim();
+        assertTrue(output.contains("Available commands:"), "Should show general help for whitespace-only input");
+    }
+
+    @Test
+    public void testHelpWithMixedCase() throws PillException {
+        HelpCommand helpCommand = new HelpCommand("AdD", false);
+        ItemMap itemMap = new ItemMap();
+        Storage storage = new Storage();
+
+        helpCommand.execute(itemMap, storage);
+
+        String output = outContent.toString().trim();
+        assertTrue(output.contains("add:"), "Should recognize command regardless of case");
+    }
+
+    @Test
+    public void testHelpWithExtraArguments() throws PillException {
+        HelpCommand helpCommand = new HelpCommand("add extra args", false);
+        ItemMap itemMap = new ItemMap();
+        Storage storage = new Storage();
+
+        helpCommand.execute(itemMap, storage);
+
+        String output = outContent.toString().trim();
+        assertTrue(output.contains("add:"), "Should ignore extra arguments and show help for 'add'");
+    }
+
+    @Test
+    public void testAllSpecificCommands() throws PillException {
+        String[] commands = {"help", "add", "delete", "edit", "list", "exit"};
+        for (String cmd : commands) {
+            outContent.reset(); // Clear the output stream
+            HelpCommand helpCommand = new HelpCommand(cmd, false);
+            ItemMap itemMap = new ItemMap();
+            Storage storage = new Storage();
+
+            helpCommand.execute(itemMap, storage);
+
+            String output = outContent.toString().trim();
+            assertTrue(output.contains(cmd + ":"), "Should show help for " + cmd + " command");
+        }
+    }
+
+    @Test
+    public void testTypoSuggestionWithNoCloseMatch() throws PillException {
+        HelpCommand helpCommand = new HelpCommand("xyz", false);
+        ItemMap itemMap = new ItemMap();
+        Storage storage = new Storage();
+
+        helpCommand.execute(itemMap, storage);
+
+        String output = outContent.toString().trim();
+        assertTrue(output.contains("Unknown command: xyz"), "Should indicate unknown command");
+        assertTrue(output.contains("Available commands:"), "Should list available commands");
+        assertFalse(output.contains("Did you mean:"), "Should not suggest a command when there's no close match");
+    }
+}
```

Comment:
Should there be more sample inputs (with varying levels of closeness to the actual command), to test the robustness of our code?